components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -66,6 +82,30 @@ interface PushConnection : Closeable {
     fun isInitialized(): Boolean
 }
 
+/**
+ * An implementation of [PushConnection] for the native component using the [PushAPI].
+ *
+ * Implementation notes: There are few important concepts to note here - for WebPush, the identifier that is required
+ * for notifying clients is called a "scope". This can be a site's host URL or an other uniquely identifying string to
+ * create a push subscription for that given scope.
+ *
+ * In the native Rust component the concept is similar, in that we have a unique ID for each subscription, however
+ * the ID is called a "channel ID" (or chid), which is a UUID. In the implementation, the scope can also be provided
+ * to the native layer, however it is mostly just stored in the database but not used in any significant way.
+ *
+ * With this in mind, we decided to write our implementation to be a 1-1 mapping of the public API 'scope' to the
+ * internal API 'channel ID' by generating a UUID from the scope value. This means that we have a reproducible way to
+ * always retrieve the chid when the caller passes the scope to us.
+ *
+ * Some nuances are that we also need to provide the native API the same scope value along with the
+ * scope-based chid value to satisfy the internal API requirements, at the cost of some noticeable duplication of
+ * information, so that we can retrieve those values later; see [RustPushConnection.subscribe] and
+ * [RustPushConnection.unsubscribe] implementations for details.
+ *
+ * Another nuance, in order to satisfy the API shape we need for push subscriptions, is to query the database for
```

Comment:
Maybe I'm not reading this right, but english seems off here. Can you reword this so it's a bit clearer? I'm not 100% what you're trying to say here.

Reply:
I swallowed some words there. Fixed! 😄